src/Pages/ProductPage/ProductPage.tsx

```diff
@@ -0,0 +1,82 @@
+import { FC, useEffect, useState } from "react";
+import { Content } from "antd/es/layout/layout";
+import { ShoppingCartOutlined } from "@ant-design/icons";
+import { Spin, Skeleton, Button, message } from "antd";
+import { useParams, useNavigate, Link } from "react-router-dom";
+import { getGoods, getGoodsLoadStatus, goodActions, cartActions, getCartGoods, getIsAuthValue } from "src/store";
+import { useSelector } from "react-redux";
+import { useAppDispatch } from "src/hooks/useAppDispatch";
+import { Good } from "src/types/general";
+import { api } from "src/api/api";
+import css from "./productPage.module.css";
+
+export const ProductPage: FC = () => {
+    const loadStatus = useSelector(getGoodsLoadStatus);
+    const isAuth = useSelector(getIsAuthValue)
+    const navigate = useNavigate();
+    const { ids } = useParams();
+    const dispatch = useAppDispatch();
+    const goods = useSelector(getGoods);
+    const cartGoods = useSelector(getCartGoods);
+    const [count, setCount] = useState<number>(1);
+    const [currentGoods, setCurrentGoods] = useState<Good | null>();
+    useEffect(() => {
+        if (!currentGoods) {
+            getGood();
```

Comment:
в useEffect если не надо переиспользовать логику, лучше вызвать код а не создавать функцию. Так мы будем видеть зависимости функции и массив зависимостей рядом. Это крайне сильно упрощает чтение и понимание кода
```suggestion
           dispatch(goodActions.serverRequest({ ids }))
```